tests/patmat/i13342.scala

```diff
@@ -0,0 +1,10 @@
+class C {
+  def m(x: true) = x match { // was: match may not be exhaustive.\nIt would fail on pattern case: false
+    case true => println("the one true path")
+  }
+
+  def n(x: true) = x match {
+    case true => 1
+    case false => 2 // was: no reachability warning on this case
```

Comment:
This continues to not be emitted, because of the hackaround that #13290 is removing in reachability analysis.

Reply:
Fixed with that merged!  (now has a patmat test checkfile)